cmd/traffic_manager/traffic_manager.cc

```diff
@@ -166,6 +166,27 @@ is_server_idle()
   return active <= threshold;
 }
 
+static bool
+is_server_draining()
+{
+  RecInt draining = 0;
+  if (RecGetRecordInt("proxy.node.config.draining", &draining) != REC_ERR_OKAY) {
+    return false;
+  }
+  return draining != 0;
+}
+
+static bool
+waited_enough()
```

Comment:
Some other way is to watch the metrics.
```
proxy.process.http2.total_client_connections
proxy.process.http.total_incoming_connections # this is only for HTTP1
```
If the incoming connections in the last second is below some configurable threshold, return true.